components/create-post-page/SetPriceAndDescription.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  Box,
+  Button,
+  Flex,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  SimpleGrid,
+  Spacer,
+  Text,
+  Textarea,
+  VStack,
+} from "@chakra-ui/react";
+import { PopulatedBook } from "@lib/services/book";
+import { ChangeEvent, FormEvent, useState } from "react";
+import BookCard from "../BookCard";
+
+interface Props {
+  book: PopulatedBook;
+  onSubmitPressed: (description: string, price: number) => void;
+}
+
+export default function SetPriceAndDescription({
+  book,
+  onSubmitPressed,
+}: Props) {
+  const [description, setDescription] = useState("");
+  const [price, setPrice] = useState(0);
+  const [valid, setValid] = useState(false);
+  const [sending, setSending] = useState(false);
+
+  const handleSubmit = (event: FormEvent) => {
+    if (!isNaN(price) && isFinite(price) && price >= 0) {
+      setSending(true);
+      onSubmitPressed(description, price);
+    }
+    event.preventDefault();
+  };
+
+  const handlePriceChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const newPriceString = e.target.value.trim();
+    const newPriceAsFloat = Number.parseFloat(newPriceString);
+    if (
+      !isNaN(newPriceAsFloat) &&
+      isFinite(newPriceAsFloat) &&
+      newPriceAsFloat >= 0
+    ) {
+      setPrice(newPriceAsFloat);
+      setValid(true);
+    } else {
+      setValid(false);
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <VStack spacing={7}>
+        <BookCard book={book} isLinkActive={false} />
+        <SimpleGrid
+          columns={2}
+          spacing={3}
+          templateColumns="1fr 4fr"
+          style={{ alignItems: "center" }}
+        >
+          {/* Row 1 */}
+          <Box>
+            <Text as="b">Description:</Text>
```

Comment:
We should use [`FormLabel`](https://chakra-ui.com/docs/form/form-control#usage) (the HTML `label` component) as it has more semantic meaning and automatically enables helpful behaviour such as clicking the label to select the input and better support for users using a screenreader. 

This will also be helpful when testing this feature with Cypress/React Testing Library (see https://testing-library.com/docs/queries/about#priority if you're interesting).